### DISCUSSION TITLE
RFC: Should privacy page be considered as a "store page"

### DISCUSSION THREAD

**Discussion by ilyasfoo** [woocommerce/woocommerce#50610] [Request for Comments]
Hey, folks!

Recently, we've allowed[^1] privacy policy page to be viewed when the store has coming soon[^2] enabled and set to `Restrict to store pages only`. This change was made because we consider privacy page to be a part of WordPress core, and not WooCommerce. An example use case is an existing WordPress blog which has a privacy policy set up, intends to start selling merchandise and installs WooCommerce. Logically, the existing privacy policy page shouldn't be hidden from public view.

The issue is we also have other parts of WooCommerce which considers the privacy policy page as a store page[^3]. I believe having different logic of what is considered as a store page is confusing and can be a facet for future bugs.

#### Solution 1: Consolidate "store page" logic 

We can consolidate the logic so privacy policy page is no longer considered a store page. This may have impact on blocks since there are some references[^4] to it, but I'm unsure to what extent. Perhaps @woocommerce/woo-fse could chime in here.

#### Solution 2: Make the situation for coming soon mode unique

It's possible that privacy policy page can considered as a store page and coming soon's use case is unique. In this case, we should differentiate the naming for the logic so instead of calling it a "store page", maybe call it something along the lines of "coming soon public page".

cc @woocommerce/ghidorah 

[^1]: Reference to [PR for allowing privacy policy page](https://github.com/woocommerce/woocommerce/pull/50608) in coming soon store-only mode.
[^2]: Reference to [coming soon mode](https://woocommerce.com/document/configuring-woocommerce-settings/coming-soon-mode) admin documentation.
[^3]: Reference to code with [privacy policy page considered as a store page](https://github.com/woocommerce/woocommerce/blob/9ff42c8b0910d562005971e025b4ef8c1333bae8/plugins/woocommerce/src/Blocks/Assets/AssetDataRegistry.php#L152).
[^4]: Reference to [`storePages` in code](https://github.com/search?q=repo%3Awoocommerce%2Fwoocommerce%20storePages&type=code)

**Comment by rjchow** [comment]
I'm with solution 2, and I think we should go even further and let admins see and modify the list of pages being walled.

**Comment by adrianduffell** [comment]
> we should go even further and let admins see and modify the list of pages being walled

There is a [filter](https://github.com/Marianne380/woocommerce/blob/19ca226649fa0802505ed2334cd8e870346de820/plugins/woocommerce/src/Admin/WCAdminHelper.php#L163-L169) to modify this, at least.

**Reply by ilyasfoo** [reply]
Good point. The filter name `woocommerce_store_pages` is going to be confusing though 😐

**Reply by ilyasfoo** [reply]
I've added extensibility docs for `woocommerce_store_pages` filter: https://github.com/woocommerce/woocommerce/pull/51160

**Comment by Aljullu** [comment]
> Reference to code with [privacy policy page considered as a store page](https://github.com/woocommerce/woocommerce/blob/9ff42c8b0910d562005971e025b4ef8c1333bae8/plugins/woocommerce/src/Blocks/Assets/AssetDataRegistry.php#L152). [↩](https://github.com/woocommerce/woocommerce/discussions/50610#user-content-fnref-3-f9f91a5c420c6498e9f818a4482ffddf)

My understanding is that even though that method is called `get_store_pages()`, that doesn't mean those are pages created by WooCommerce. Instead, they are pages which are needed by some blocks (ie, to add a link to them, see https://github.com/woocommerce/woocommerce-blocks/pull/1881). I understand how it can be confusing so it might be a good idea to add a comment explaining that or even change the function name.

**Reply by ilyasfoo** [reply]
Thanks for the feedback! Changing the function name would be great if it's not too complicated - I imagine the biggest risk is if there's extensibility or if it's a commonly used exported function.

**Reply by Aljullu** [reply]
Makes sense! Issue created: https://github.com/woocommerce/woocommerce/issues/50658.
